spec/datadog/appsec/component_spec.rb

```diff
@@ -19,6 +19,20 @@
         expect(component).to be_a(described_class)
       end
 
+      context 'when using old ffi version with Ruby 3.3.x' do
+        before do
+          stub_const('RUBY_VERSION', '3.3.0')
+          allow(Gem).to receive(:loaded_specs).and_return('ffi' => double(version: Gem::Version.new('1.15.4')))
```

Comment:
If you use 1.9.0 for the FFI version the test will fail right?

Reply:
no, it is still green:

```ruby
Gem::Version.new('1.9.0') < '1.16.0'
=> true
```

I think this is because we are comparing `Gem::Version` with a `String` (not sure this will work correctly in older ruby versions though)

Reply:
We can dismantle version on major, minor and test piece-by-piece (just in case)

Reply:
Yep, https://github.com/rubygems/rubygems/blob/master/lib/rubygems/version.rb#L358, added in https://github.com/rubygems/rubygems/commit/7e0dbb79f2c351532ae6c8cf12d5b39579dc4c54 2 years ago, which sounds like this functionality is probably not going to exist in Ruby 2.5?

Reply:
https://github.com/rails/rails/pull/47480 conveniently says that the threshold for this feature is Ruby 3.1.